packages/core/src/Modules/Cashier/Containers/p2p-cashier.jsx

```diff
@@ -1,21 +1,53 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
 import P2P from '@deriv/p2p';
 import { getLanguage } from '@deriv/translations';
 import { WS } from 'Services';
 import { connect } from 'Stores/connect';
 import ServerTime from '_common/base/server_time';
 
 /* P2P will use the same websocket connection as Deriv/Binary, we need to pass it as a prop */
-const P2PCashier = ({ currency, local_currency_config, is_virtual, residence, setNotificationCount }) => (
-    <P2P
-        websocket_api={WS}
-        lang={getLanguage()}
-        client={{ currency, local_currency_config, is_virtual, residence }}
-        server_time={ServerTime}
-        setNotificationCount={setNotificationCount}
-    />
-);
+class P2PCashier extends React.Component {
+    state = {
+        order_id: null,
+    };
+
+    setQueryOrder = input_order_id => {
+        const { history, location } = this.props;
+        const order_id_param = input_order_id ? `?order=${input_order_id}` : '';
+
+        if (this.state.order_id !== input_order_id) {
+            // Changing query params
+            history.push({ pathname: '/cashier/p2p', search: order_id_param, hash: location.hash });
```

Comment:
Has this been tested? I think we'd need to use `replace` instead of `push`. And also, when adding the search and hash back, it should remember to maintain any other search params, not drop the old ones.

Reply:
yes, i tested it, it works, ya you're right, i need to put the previous search param as well, thanks